File: composeApp/src/androidMain/kotlin/com/po4yka/trailglass/ui/screens/TripDetailScreen.kt
package com.po4yka.trailglass.ui.screens

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.po4yka.trailglass.domain.model.Trip
import kotlinx.datetime.TimeZone
import kotlinx.datetime.toLocalDateTime
import kotlin.time.Duration.Companion.hours

/**
 * Detailed view of a single trip with statistics, timeline, and actions.
 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun TripDetailScreen(
    trip: Trip,
    onBack: () -> Unit = {},
    onEdit: () -> Unit = {},
    onShare: () -> Unit = {},
    onExport: (ExportFormat) -> Unit = {},
    onDelete: () -> Unit = {},
    modifier: Modifier = Modifier
) {
    var showExportMenu by remember { mutableStateOf(false) }
    var showDeleteDialog by remember { mutableStateOf(false) }

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text(trip.displayName) },
                navigationIcon = {
                    IconButton(onClick = onBack) {
                        Icon(Icons.Default.ArrowBack, contentDescription = "Back")
                    }
                },
                actions = {
                    IconButton(onClick = onShare) {
                        Icon(Icons.Default.Share, contentDescription = "Share")
                    }
                    IconButton(onClick = { showExportMenu = true }) {
                        Icon(Icons.Default.Download, contentDescription = "Export")
                    }
                    IconButton(onClick = onEdit) {
                        Icon(Icons.Default.Edit, contentDescription = "Edit")
                    }
                    IconButton(onClick = { showDeleteDialog = true }) {
                        Icon(Icons.Default.Delete, contentDescription = "Delete")
                    }

                    // Export menu
                    DropdownMenu(
                        expanded = showExportMenu,
                        onDismissRequest = { showExportMenu = false }
                    ) {
                        DropdownMenuItem(
                            text = { Text("Export as GPX") },
                            onClick = {
                                showExportMenu = false
                                onExport(ExportFormat.GPX)
                            },
                            leadingIcon = {
                                Icon(Icons.Default.Map, contentDescription = null)
                            }
                        )
                        DropdownMenuItem(
                            text = { Text("Export as KML") },
                            onClick = {
                                showExportMenu = false
                                onExport(ExportFormat.KML)
                            },
                            leadingIcon = {
                                Icon(Icons.Default.Public, contentDescription = null)
                            }
                        )
                    }
                }
            )
        },
        modifier = modifier.fillMaxSize()
    ) { paddingValues ->
        LazyColumn(
            modifier = Modifier.padding(paddingValues),
            contentPadding = PaddingValues(16.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            // Trip header card
            item {
                TripHeaderCard(trip)
            }

            // Statistics card
            item {
                TripStatisticsCard(trip)
            }

            // Description card
            if (trip.description != null) {
                item {
                    Card(modifier = Modifier.fillMaxWidth()) {
                        Column(modifier = Modifier.padding(16.dp)) {
                            Text(
                                text = "Description",
                                style = MaterialTheme.typography.titleMedium,
                                fontWeight = FontWeight.Bold
                            )
                            Spacer(modifier = Modifier.height(8.dp))
                            Text(
                                text = trip.description!!,
                                style = MaterialTheme.typography.bodyMedium
                            )
                        }
                    }
                }
            }

            // Tags
            if (trip.tags.isNotEmpty()) {
                item {
                    Card(modifier = Modifier.fillMaxWidth()) {
                        Column(modifier = Modifier.padding(16.dp)) {
                            Text(
                                text = "Tags",
                                style = MaterialTheme.typography.titleMedium,
                                fontWeight = FontWeight.Bold
                            )
                            Spacer(modifier = Modifier.height(8.dp))
                            Row(
                                horizontalArrangement = Arrangement.spacedBy(8.dp),
                                modifier = Modifier.fillMaxWidth()
                            ) {
                                trip.tags.forEach { tag ->
                                    AssistChip(
                                        onClick = { },
                                        label = { Text(tag) },
                                        leadingIcon = {
                                            Icon(
                                                Icons.Default.Tag,
                                                contentDescription = null,
                                                modifier = Modifier.size(16.dp)
                                            )
                                        }
                                    )
                                }
                            }
                        }
                    }
                }
            }

            // Visited countries and cities
            if (trip.countriesVisited.isNotEmpty() || trip.citiesVisited.isNotEmpty()) {
                item {
                    VisitedPlacesCard(
                        countries = trip.countriesVisited,
                        cities = trip.citiesVisited
                    )
                }
            }
        }
    }

    // Delete confirmation dialog
    if (showDeleteDialog) {
        AlertDialog(
            onDismissRequest = { showDeleteDialog = false },
            title = { Text("Delete Trip?") },
            text = { Text("This action cannot be undone. All trip data will be permanently deleted.") },
            confirmButton = {
                TextButton(
                    onClick = {
                        showDeleteDialog = false
                        onDelete()
                    },
                    colors = ButtonDefaults.textButtonColors(
                        contentColor = MaterialTheme.colorScheme.error
                    )
                ) {
                    Text("Delete")
                }
            },
            dismissButton = {
                TextButton(onClick = { showDeleteDialog = false }) {
                    Text("Cancel")
                }
            }
        )
    }
}

@Composable
private fun TripHeaderCard(trip: Trip) {
    Card(
        modifier = Modifier.fillMaxWidth(),
        colors = CardDefaults.cardColors(
            containerColor = if (trip.isOngoing)
                MaterialTheme.colorScheme.primaryContainer
            else
                MaterialTheme.colorScheme.surfaceVariant
        )
    ) {
        Column(modifier = Modifier.padding(16.dp)) {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.Top
            ) {
                Column(modifier = Modifier.weight(1f)) {
                    Text(
                        text = trip.displayName,
                        style = MaterialTheme.typography.headlineMedium,
                        fontWeight = FontWeight.Bold
                    )

                    Spacer(modifier = Modifier.height(8.dp))

                    // Date range
                    val startDate = trip.startTime.toLocalDateTime(TimeZone.currentSystemDefault())
                    val dateText = if (trip.endTime != null) {
                        val endDate = trip.endTime.toLocalDateTime(TimeZone.currentSystemDefault())
                        "${startDate.date} to ${endDate.date}"
                    } else {
                        "Started ${startDate.date}"
                    }

                    Row(
                        horizontalArrangement = Arrangement.spacedBy(4.dp),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Icon(
                            Icons.Default.CalendarToday,
                            contentDescription = null,
                            modifier = Modifier.size(16.dp)
                        )
                        Text(
                            text = dateText,
                            style = MaterialTheme.typography.bodyMedium
                        )
                    }

                    // Duration
                    if (trip.duration != null) {
                        Spacer(modifier = Modifier.height(4.dp))
                        Row(
                            horizontalArrangement = Arrangement.spacedBy(4.dp),
                            verticalAlignment = Alignment.CenterVertically
                        ) {
                            Icon(
                                Icons.Default.Schedule,
                                contentDescription = null,
                                modifier = Modifier.size(16.dp)
                            )
                            Text(
                                text = formatDuration(trip.duration!!),
                                style = MaterialTheme.typography.bodyMedium
                            )
                        }
                    }
                }

                // Badges
                Column(horizontalAlignment = Alignment.End) {
                    if (trip.isOngoing) {
                        Badge(
                            containerColor = MaterialTheme.colorScheme.primary
                        ) {
                            Text("Ongoing")
                        }
                    }
                    if (trip.isAutoDetected) {
                        Spacer(modifier = Modifier.height(8.dp))
                        Badge(
                            containerColor = MaterialTheme.colorScheme.tertiaryContainer
                        ) {
                            Text("Auto-detected")
                        }
                    }
                    if (trip.isPublic) {
                        Spacer(modifier = Modifier.height(8.dp))
                        Badge {
                            Text("Public")
                        }
                    }
                }
            }
        }
    }
}

@Composable
private fun TripStatisticsCard(trip: Trip) {
    Card(modifier = Modifier.fillMaxWidth()) {
        Column(modifier = Modifier.padding(16.dp)) {
            Text(
                text = "Statistics",
                style = MaterialTheme.typography.titleMedium,
                fontWeight = FontWeight.Bold
            )

            Spacer(modifier = Modifier.height(12.dp))

            // Statistics grid
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceEvenly
            ) {
                if (trip.totalDistanceMeters > 0) {
                    StatItem(
                        icon = Icons.Default.Straighten,
                        label = "Distance",
                        value = "${(trip.totalDistanceMeters / 1000).toInt()} km"
                    )
                }
                if (trip.visitedPlaceCount > 0) {
                    StatItem(
                        icon = Icons.Default.Place,
                        label = "Places",
                        value = "${trip.visitedPlaceCount}"
                    )
                }
                if (trip.countriesVisited.isNotEmpty()) {
                    StatItem(
                        icon = Icons.Default.Public,
                        label = "Countries",
                        value = "${trip.countriesVisited.size}"
                    )
                }
            }
        }
    }
}

@Composable
private fun StatItem(
    icon: androidx.compose.ui.graphics.vector.ImageVector,
    label: String,
    value: String
) {
    Column(horizontalAlignment = Alignment.CenterHorizontally) {
        Icon(
            icon,
            contentDescription = null,
            modifier = Modifier.size(24.dp),
            tint = MaterialTheme.colorScheme.primary
        )
        Spacer(modifier = Modifier.height(4.dp))
        Text(
            text = value,
            style = MaterialTheme.typography.titleMedium,
            fontWeight = FontWeight.Bold
        )
        Text(
            text = label,
            style = MaterialTheme.typography.bodySmall,
            color = MaterialTheme.colorScheme.onSurfaceVariant
        )
    }
}

@Composable
private fun VisitedPlacesCard(
    countries: List<String>,
    cities: List<String>
) {
    Card(modifier = Modifier.fillMaxWidth()) {
        Column(modifier = Modifier.padding(16.dp)) {
            if (countries.isNotEmpty()) {
                Text(
                    text = "Countries Visited",
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Bold
                )
                Spacer(modifier = Modifier.height(8.dp))
                Text(
                    text = countries.joinToString(", "),
                    style = MaterialTheme.typography.bodyMedium
                )
            }

            if (cities.isNotEmpty()) {
                if (countries.isNotEmpty()) {
                    Spacer(modifier = Modifier.height(16.dp))
                }
                Text(
                    text = "Cities Visited",
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Bold
                )
                Spacer(modifier = Modifier.height(8.dp))
                Text(
                    text = cities.take(10).joinToString(", ") +
                            if (cities.size > 10) " and ${cities.size - 10} more" else "",
                    style = MaterialTheme.typography.bodyMedium
                )
            }
        }
    }
}

enum class ExportFormat {
    GPX, KML
}

private fun formatDuration(duration: kotlin.time.Duration): String {
    val days = duration.inWholeDays
    val hours = duration.inWholeHours % 24
    val minutes = duration.inWholeMinutes % 60

    return when {
        days > 0 && hours > 0 -> "$days days, $hours hours"
        days > 0 -> "$days days"
        hours > 0 && minutes > 0 -> "$hours hours, $minutes min"
        hours > 0 -> "$hours hours"
        else -> "$minutes min"
    }
}

File: iosApp/iosApp/Screens/TripDetailScreen.swift
import SwiftUI
import Shared

/**
 * SwiftUI trip detail screen matching Android TripDetailScreen.
 * Shows detailed view of a single trip with statistics, timeline, and actions.
 */
struct TripDetailView: View {
    let trip: Trip
    let onBack: () -> Void
    let onEdit: () -> Void
    let onShare: () -> Void
    let onExport: (ExportFormat) -> Void
    let onDelete: () -> Void

    @State private var showExportMenu = false
    @State private var showDeleteDialog = false

    var body: some View {
        NavigationView {
            ScrollView {
                LazyVStack(spacing: 16) {
                    // Trip header card
                    TripHeaderCard(trip: trip)

                    // Statistics card
                    TripStatisticsCard(trip: trip)

                    // Description card
                    if !trip.description.isEmpty {
                        Card {
                            VStack(alignment: .leading, spacing: 8) {
                                Text("Description")
                                    .font(.headline)

                                Text(trip.description)
                                    .font(.body)
                            }
                            .padding(16)
                        }
                    }

                    // Tags
                    if !trip.tags.isEmpty {
                        Card {
                            VStack(alignment: .leading, spacing: 8) {
                                Text("Tags")
                                    .font(.headline)

                                HStack(spacing: 8) {
                                    ForEach(trip.tags, id: \.self) { tag in
                                        HStack(spacing: 4) {
                                            Image(systemName: "tag")
                                                .font(.caption)
                                            Text(tag)
                                                .font(.body)
                                        }
                                        .padding(.horizontal, 8)
                                        .padding(.vertical, 4)
                                        .background(Color(.systemGray5))
                                        .cornerRadius(4)
                                    }
                                }
                            }
                            .padding(16)
                        }
                    }

                    // Visited countries and cities
                    if !trip.countriesVisited.isEmpty || !trip.citiesVisited.isEmpty {
                        VisitedPlacesCard(
                            countries: trip.countriesVisited,
                            cities: trip.citiesVisited
                        )
                    }
                }
                .padding(16)
            }
            .navigationTitle(trip.displayName)
            .navigationBarTitleDisplayMode(.inline)
            .toolbar {
                ToolbarItem(placement: .navigationBarLeading) {
                    Button(action: onBack) {
                        Image(systemName: "chevron.left")
                    }
                }

                ToolbarItem(placement: .navigationBarTrailing) {
                    Menu {
                        Button(action: onShare) {
                            Label("Share", systemImage: "square.and.arrow.up")
                        }

                        Menu {
                            Button(action: { onExport(.gpx) }) {
                                Label("Export as GPX", systemImage: "map")
                            }
                            Button(action: { onExport(.kml) }) {
                                Label("Export as KML", systemImage: "globe")
                            }
                        } label: {
                            Label("Export", systemImage: "arrow.down.doc")
                        }

                        Button(action: onEdit) {
                            Label("Edit", systemImage: "pencil")
                        }

                        Divider()

                        Button(role: .destructive, action: { showDeleteDialog = true }) {
                            Label("Delete", systemImage: "trash")
                        }
                    } label: {
                        Image(systemName: "ellipsis.circle")
                    }
                }
            }
            .alert("Delete Trip?", isPresented: $showDeleteDialog) {
                Button("Cancel", role: .cancel) { }
                Button("Delete", role: .destructive, action: onDelete)
            } message: {
                Text("This action cannot be undone. All trip data will be permanently deleted.")
            }
        }
    }
}

/**
 * Trip header card.
 */
private struct TripHeaderCard: View {
    let trip: Trip

    var body: some View {
        Card {
            VStack(alignment: .leading, spacing: 12) {
                HStack(alignment: .top) {
                    VStack(alignment: .leading, spacing: 8) {
                        Text(trip.displayName)
                            .font(.title2)
                            .fontWeight(.bold)

                        // Date range
                        HStack(spacing: 4) {
                            Image(systemName: "calendar")
                                .font(.caption)
                            Text(formatDateRange(trip))
                                .font(.body)
                        }

                        // Duration
                        if let duration = trip.duration as? KotlinDuration {
                            HStack(spacing: 4) {
                                Image(systemName: "clock")
                                    .font(.caption)
                                Text(formatDuration(duration))
                                    .font(.body)
                            }
                        }
                    }

                    Spacer()

                    // Badges
                    VStack(alignment: .trailing, spacing: 8) {
                        if trip.isOngoing {
                            Text("Ongoing")
                                .font(.caption)
                                .padding(.horizontal, 8)
                                .padding(.vertical, 4)
                                .background(Color.blue)
                                .foregroundColor(.white)
                                .cornerRadius(4)
                        }

                        if trip.isAutoDetected {
                            Text("Auto-detected")
                                .font(.caption)
                                .padding(.horizontal, 8)
                                .padding(.vertical, 4)
                                .background(Color.purple.opacity(0.2))
                                .foregroundColor(.purple)
                                .cornerRadius(4)
                        }

                        if trip.isPublic {
                            Text("Public")
                                .font(.caption)
                                .padding(.horizontal, 8)
                                .padding(.vertical, 4)
                                .background(Color.green.opacity(0.2))
                                .foregroundColor(.green)
                                .cornerRadius(4)
                        }
                    }
                }
            }
            .padding(16)
        }
        .background(trip.isOngoing ? Color.blue.opacity(0.1) : Color(.systemGray6))
        .cornerRadius(12)
    }

    private func formatDateRange(_ trip: Trip) -> String {
        let dateFormatter = DateFormatter()
        dateFormatter.dateStyle = .medium
        dateFormatter.timeStyle = .none

        let startDate = dateFormatter.string(from: Date(timeIntervalSince1970: Double(trip.startTime.epochSeconds)))

        if let endTime = trip.endTime {
            let endDate = dateFormatter.string(from: Date(timeIntervalSince1970: Double(endTime.epochSeconds)))
            return "\(startDate) to \(endDate)"
        } else {
            return "Started \(startDate)"
        }
    }

    private func formatDuration(_ duration: KotlinDuration) -> String {
        let totalHours = duration.inWholeHours
        let days = totalHours / 24
        let hours = totalHours % 24
        let minutes = duration.inWholeMinutes % 60

        if days > 0 && hours > 0 {
            return "\(days) days, \(hours) hours"
        } else if days > 0 {
            return "\(days) days"
        } else if hours > 0 && minutes > 0 {
            return "\(hours) hours, \(minutes) min"
        } else if hours > 0 {
            return "\(hours) hours"
        } else {
            return "\(minutes) min"
        }
    }
}

/**
 * Trip statistics card.
 */
private struct TripStatisticsCard: View {
    let trip: Trip

    var body: some View {
        Card {
            VStack(alignment: .leading, spacing: 12) {
                Text("Statistics")
                    .font(.headline)

                HStack(spacing: 16) {
                    if trip.totalDistanceMeters > 0 {
                        StatItem(
                            icon: "ruler",
                            label: "Distance",
                            value: "\(Int(trip.totalDistanceMeters / 1000)) km"
                        )
                    }

                    if trip.visitedPlaceCount > 0 {
                        StatItem(
                            icon: "mappin.and.ellipse",
                            label: "Places",
                            value: "\(trip.visitedPlaceCount)"
                        )
                    }

                    if !trip.countriesVisited.isEmpty {
                        StatItem(
                            icon: "globe",
                            label: "Countries",
                            value: "\(trip.countriesVisited.count)"
                        )
                    }
                }
                .frame(maxWidth: .infinity)
            }
            .padding(16)
        }
    }
}

/**
 * Stat item.
 */
private struct StatItem: View {
    let icon: String
    let label: String
    let value: String

    var body: some View {
        VStack(spacing: 4) {
            Image(systemName: icon)
                .font(.title3)
                .foregroundColor(.blue)

            Text(value)
                .font(.headline)
                .fontWeight(.bold)

            Text(label)
                .font(.caption)
                .foregroundColor(.secondary)
        }
    }
}

/**
 * Visited places card.
 */
private struct VisitedPlacesCard: View {
    let countries: [String]
    let cities: [String]

    var body: some View {
        Card {
            VStack(alignment: .leading, spacing: 16) {
                if !countries.isEmpty {
                    VStack(alignment: .leading, spacing: 8) {
                        Text("Countries Visited")
                            .font(.headline)

                        Text(countries.joined(separator: ", "))
                            .font(.body)
                    }
                }

                if !cities.isEmpty {
                    VStack(alignment: .leading, spacing: 8) {
                        Text("Cities Visited")
                            .font(.headline)

                        let cityText = cities.prefix(10).joined(separator: ", ")
                            + (cities.count > 10 ? " and \(cities.count - 10) more" : "")

                        Text(cityText)
                            .font(.body)
                    }
                }
            }
            .padding(16)
        }
    }
}

/**
 * Reusable card component.
 */
private struct Card<Content: View>: View {
    let content: Content

    init(@ViewBuilder content: () -> Content) {
        self.content = content()
    }

    var body: some View {
        content
            .frame(maxWidth: .infinity, alignment: .leading)
            .background(Color(.systemBackground))
            .cornerRadius(12)
            .shadow(radius: 2)
    }
}

/**
 * Export format enum.
 */
enum ExportFormat {
    case gpx
    case kml
}
